### PR TITLE
doc: update documentation

### PR DESCRIPTION
**Pull request by maurocchi**
Updated the library documentation, generated with `mix docs`.

### CODE REVIEW

File: guides/unreachable-analyzer.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+* `report_transitively_unused`: if true the analyzer reports also the unused 
+  functions that are only called by other unused functions (default to false).
+
+  For instance, if `a()` calls `b()` that calls `c()`, by default only `a()` is reported as unused. Additionally, if `a()` is explicitly ignored or is defined out of the configured `paths`, nothing is reported.
+
+## Corner cases
+
+* Behaviours implementers are not considered used by default.
+* Functions generated by a macro are not reported, since they generally are out
+  of the our control.
```
Comment (korialis) [inline guides/unreachable-analyzer.md]:
```suggestion
  of our control.
```

File: guides/unreachable-analyzer.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        usages_discovery: [
+          MyDiscovery,
+          # ...
+        ]
+      }}
+  ]
+]
+```
+
+A discovery module implements the [MixUnused.Analyzers.Unreachable.Usages](`MixUnused.Analyzers.Unreachable.Usages`) behaviour and it's called during the analysis to try to discover functions that should considered as used.
```
Comment (korialis) [inline guides/unreachable-analyzer.md]:
```suggestion
A discovery module implements the [MixUnused.Analyzers.Unreachable.Usages](`MixUnused.Analyzers.Unreachable.Usages`) behaviour and it's called during the analysis to try to discover functions that should be considered as used.
```

File: guides/unreachable-analyzer.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+The [MixUnused.Analyzers.Unreachable](`MixUnused.Analyzers.Unreachable`) check analyses the call graph by visiting all the functions reachable from a well-known set of functions, that act as starting points.
+
+All the remaining functions are considered unused, and by default only the functions without any caller are reported to the user.
```
Comment (korialis) [inline guides/unreachable-analyzer.md]:
```suggestion
All the non reachable functions are considered unused. By default only the functions that have no caller are reported to the user.
```

File: lib/mix_unused/analyzers/unreachable/config.ex [hunk 4]
```diff
@@ -1,5 +1,7 @@
 defmodule MixUnused.Analyzers.Unreachable.Config do
-  @moduledoc false
+  @moduledoc """
+  Configuration specific for the [Unreachable](`MixUnused.Analyzers.Unreachable`) analyzer.
```
Comment (korialis) [inline lib/mix_unused/analyzers/unreachable/config.ex]:
```suggestion
  Configuration specific to the [Unreachable](`MixUnused.Analyzers.Unreachable`) analyzer.
```

File: lib/mix_unused/analyzers/unreachable/usages/absinthe_discovery.ex [hunk 5]
```diff
@@ -1,6 +1,13 @@
 defmodule MixUnused.Analyzers.Unreachable.Usages.AbsintheDiscovery do
   @moduledoc """
-  Discovers the modules used by [Absinthe](https://hex.pm/packages/absinthe).
+  Discovers some components used dynamically by the [Absinthe](http://absinthe-graphql.org/).
```
Comment (korialis) [inline lib/mix_unused/analyzers/unreachable/usages/absinthe_discovery.ex]:
```suggestion
  Discovers some components used dynamically by [Absinthe](http://absinthe-graphql.org/).
```

File: lib/mix_unused/analyzers/unreachable/usages/supervisor_discovery.ex [hunk 6]
```diff
@@ -1,6 +1,33 @@
 defmodule MixUnused.Analyzers.Unreachable.Usages.SupervisorDiscovery do
   @moduledoc """
   Discovers the GenServers started by Supervisor.
+
+  The current implementation is quite naive: it looks at any file named _supervisor.ex_, searches all the referred modules and considers as used all the functions related with the Genserver behaviour that exists in the current application.
```
Comment (korialis) [inline lib/mix_unused/analyzers/unreachable/usages/supervisor_discovery.ex]: I am not sure about this, but I feel that a rephrase would make it clearer:

> searches all the referred modules and considers as used all the functions related with the Genserver behaviour that exist in the referred modules.

**Comment by korialis** [comment]
Looks good, left a few comments with minor suggestions

**Comment by maurocchi** [comment]
Thanks @korialis!

**Pull request merged**